TrappingRainWater/main.cpp

```diff
@@ -0,0 +1,75 @@
+#include "main.h"
+
+
+int TrappingRainWaterSolution::trap(std::span<const int> height) {
+  // Get the size of the elevation map
+  size_t n = height.size();
+
+  // Edge case: can't trap water with single element
+  if (n == 1) {
+    return 0;
+  }
+
+  // lp (left pointer) and rp (right pointer) track boundaries of water sections
+  int lp = n;
+  int rp = 0;
+  int total = 0;
+
+  // First pass: scan left to right looking for trapped water
+  for (int i = 0; i < n; i++) {
```

Comment:
what's the complexity? I see two nested loops, so looks like O(N^2)
Do you think you can find a better one?